src/controlplane/client/client.go

```diff
@@ -0,0 +1,59 @@
+package client
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/newrelic/infra-integrations-sdk/log"
+
+	"github.com/newrelic/nri-kubernetes/v2/src/client"
+	"github.com/newrelic/nri-kubernetes/v2/src/prometheus"
+)
+
+type Client struct {
+	logger           log.Logger
+	connectionParams *connParams
+}
+
+type Config struct {
+	Logger    log.Logger
+	Connector Connector
```

Comment:
I don't understand why the connector is decoupled, could there be other connectors? Wouldn't it make more sense to pass all the connector config here and make the client construct the Default connector in the New? As I said, maybe I am loosing something and there can be other connectors.

Reply:
In the kubelet it was embedded, in the first iteration, in the client itself and we ended up extracting it to reduce its logic keeping it "clean". 😃 
Right now the only other connected that exists is the fixedConnector, used in tests only.

Reply:
oks, I just saw the place where the client is created is really long with the population of the connector config, the creation of the connector, then the population of the client config and the client creation, and just wondered if it could be done differently. One possibility is to have the Client config accept the defaultConnector parameters, plus a Connector pointer, if the connector is nil, then the DefaultConnector is created inside the client's new, if not the Connector passed is used. But really, it's not important, it's just a matter of taste.